workspace/workspace.go

```diff
@@ -102,3 +103,53 @@ func (ws Workspace) Locate(exercise string) ([]string, error) {
 	}
 	return paths, nil
 }
+
+func (ws Workspace) SolutionPath(exercise, solutionID string) (string, error) {
+	paths, err := ws.Locate(exercise)
+	if !IsNotExist(err) && err != nil {
+		return "", err
+	}
+
+	return ws.DetermineSolutionPath(paths, exercise, solutionID, PathContainsSolution)
+}
+
+// TODO: PathContainsSolution might need a different name.
```

Comment:
I like `IsSolutionDir` and `IsSolutionPath`. I'm thinking _path_ rather than _dir_ since we're passing in an absolute path.

If we go with one of those then we'll need to swap the order of the arguments. My thinking is:

- **path** contains **solution** -> path, solution
- is **solution** **path** -> solution, path